swig/csharp/CMakeLists.txt

```diff
@@ -9,275 +15,753 @@ set(GDAL_CSHARP_INSTALL_DIR
+set_property(TEST csharp_osrtransform PROPERTY ENVIRONMENT "${TEST_ENV}")
+add_test(
+  NAME csharp_gdalwrite
+  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+  COMMAND ${CSHARP_INTERPRETER} GDALWrite/GDALWrite Data/sample.tif)
+set_property(TEST csharp_gdalwrite PROPERTY ENVIRONMENT "${TEST_ENV}")
+set_property(TEST csharp_gdalwrite PROPERTY DEPENDS csharp_createdata)
+add_test(
+  NAME csharp_gdaldatasetwrite
+  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+  COMMAND ${CSHARP_INTERPRETER} GDALDatasetWrite/GDALDatasetWrite Data/sample1.tif)
+set_property(TEST csharp_gdaldatasetwrite PROPERTY ENVIRONMENT "${TEST_ENV}")
+set_property(TEST csharp_gdaldatasetwrite PROPERTY DEPENDS csharp_createdata)
+add_test(
+  NAME csharp_gdalcreatecopy
+  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+  COMMAND ${CSHARP_INTERPRETER} GDALCreateCopy/GDALCreateCopy Data/sample.tif Data/sample2.tif)
+set_property(TEST csharp_gdalcreatecopy PROPERTY ENVIRONMENT "${TEST_ENV}")
+set_property(TEST csharp_gdalcreatecopy PROPERTY DEPENDS csharp_createdata)
+add_test(
+  NAME csharp_gdalreaddirect
+  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+  COMMAND ${CSHARP_INTERPRETER} GDALReadDirect/GDALReadDirect Data/sample.tif Data/sample.png)
+set_property(TEST csharp_gdalreaddirect PROPERTY ENVIRONMENT "${TEST_ENV}")
+set_property(TEST csharp_gdalreaddirect PROPERTY DEPENDS csharp_createdata)
+add_test(
+  NAME csharp_gdaloverviews
+  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+  COMMAND ${CSHARP_INTERPRETER} GDALOverviews/GDALOverviews Data/sample.tif NEAREST 2 4)
+set_property(TEST csharp_gdaloverviews PROPERTY ENVIRONMENT "${TEST_ENV}")
+set_property(TEST csharp_gdaloverviews PROPERTY DEPENDS csharp_createdata)
+add_test(
+  NAME csharp_gdalinfo
+  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+  COMMAND ${CSHARP_INTERPRETER} GDALInfo/GDALInfo Data/sample.tif)
+set_property(TEST csharp_gdalinfo PROPERTY ENVIRONMENT "${TEST_ENV}")
+set_property(TEST csharp_gdalinfo PROPERTY DEPENDS csharp_createdata)
+add_test(
+  NAME csharp_ogrlayeralg
+  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+  COMMAND ${CSHARP_INTERPRETER} OGRLayerAlg/OGRLayerAlg Intersection Data/pointlayer.shp pointlayer Data/pointlayer.shp
+          pointlayer Data intersectionresult)
+set_property(TEST csharp_ogrlayeralg PROPERTY ENVIRONMENT "${TEST_ENV}")
+set_property(TEST csharp_ogrlayeralg PROPERTY DEPENDS csharp_createdata)
+add_test(
+  NAME csharp_ogrlayeralgresult
+  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+  COMMAND ${CSHARP_INTERPRETER} OgrInfo/OgrInfo Data/intersectionresult.shp)
+set_property(TEST csharp_ogrlayeralgresult PROPERTY ENVIRONMENT "${TEST_ENV}")
+set_property(TEST csharp_ogrlayeralgresult PROPERTY DEPENDS csharp_ogrlayeralg)
+add_test(
+  NAME csharp_gdalwarp
+  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+  COMMAND ${CSHARP_INTERPRETER} GDALWarp/GDALWarp Data/testwarp.tif "-of GTiff -t_srs EPSG:32645 -overwrite"
+          "../../autotest/gcore/data/byte.tif")
```

Comment:
if people build from a tarball, the autotest directory will not be present. You could just copy byte.tif into the swig/csharp/Data directory.  EPSG:32645 is not super relevant for that dataset. EPSG:32611 would be better. And the -overwrite flag is only used for the gdalwarp binary, not for the gdal.Warp() API

Reply:
I think that might be a deeper question. Given the way that CMAKE works - when you build the bindings standalone although it is only building the csharp targets, it is still generating all targets and all tests. So - if you build from a tarball - I don't actually think you can just run a naked  `ctest` anyway - I think you would have to only run the individual tests you want anyway - e.g. `ctest -R "^csharp.*"`. My opinion is - since building stand-alone is not the core workflow - this should be documented and possibly raised as an issue for more investigation but should not be a blocker.

As for the details of the warp function - this is just what we used before. I would guess that it was created a number of years ago originally :) 

I would suggest - at least for the time being - that we could remove this test. I feel we have enough other tests and it would avoid any problems with this test.

Reply:
> I would suggest - at least for the time being - that we could remove this test. I feel we have enough other tests and it would avoid any problems with this test.

Fine for me. Or just comment it out with a note regarding the reason

Reply:
I commented out the test and then changed the test standalone build that builds on a conda libgdal distribution extended to run the ctest tests successfully